haystack/modeling/model/prediction_head.py

```diff
@@ -971,7 +971,9 @@ def get_similarity_function(self):
                 f"The similarity function can only be 'dot_product' or 'cosine', not '{self.similarity_function}'"
             )
 
-    def forward(self, query_vectors: torch.Tensor, passage_vectors: torch.Tensor) -> Tuple[torch.Tensor, torch.Tensor]:
+    def forward(
+        self, query_vectors: torch.Tensor, passage_vectors: Optional[torch.Tensor] = None
```

Comment:
Can you maybe explain why this change is needed?

Reply:
I made this change because I discovered that if a `BiAdaptiveModel` is instantiated with a `TextSimilarityHead` as one of its `prediction_heads`, the head can be called with `None` as the second forward parameter here: 

https://github.com/deepset-ai/haystack/blob/dc48c444d42877065517b7f316e45c6f345356a7/haystack/modeling/model/biadaptive_model.py#L299

I added the default parameter (`= None`) because it used to be necessary for my `BiAdaptiveSharedModel`. 

However, as it doesn't really have anything to do with this PR anymore, I reverted the change now 😊